app/daemon.go

```diff
@@ -84,11 +84,22 @@ type MenderShellDaemon struct {
 	homeDir                 string
 	shellsSpawned           uint
 	debug                   bool
+	router                  session.Router
 }
 
 func NewDaemon(config *configuration.MenderShellConfig) *MenderShellDaemon {
 	ctx, ctxCancel := context.WithCancel(context.Background())
 
+	routes := make(session.ProtoRoutes)
+	if config.FileTransfer.Enable {
+		routes[ws.ProtoTypeFileTransfer] = session.FileTransfer()
+	}
+	router := session.NewRouter(
+		routes, session.Config{
+			IdleTimeout: connectionmanager.DefaultPingWait,
+		},
+	)
+
```

Comment:
this needs explaining in the comments.

Reply:
Yup, I forgot to put comments explaining this, but I did a modest attempt in the commit messages.